yarn-project/sequencer-client/src/slasher/slasher_client.ts

```diff
@@ -150,31 +150,18 @@ export class SlasherClient extends WithTracer {
     return EthAddress.fromString(payloadAddress);
   }
 
-  public handleBlockStreamEvent(event: L2BlockSourceEvent): Promise<void> {
-    this.log.debug(`Handling block stream event ${event.type}`);
-    switch (event.type) {
-      case L2BlockSourceEvents.L2PruneDetected:
-        this.handlePruneL2Blocks(event);
-        break;
-      default: {
-        break;
-      }
-    }
-    return Promise.resolve();
-  }
-
   /**
    * Allows consumers to stop the instance of the slasher client.
    * 'ready' will now return 'false' and the running promise that keeps the client synced is interrupted.
    */
   public stop() {
     this.log.debug('Stopping Slasher client...');
-    this.l2BlockSource.removeListener(L2BlockSourceEvents.L2PruneDetected, this.handlePruneL2Blocks.bind(this));
+    this.l2BlockSource.removeListener(L2BlockSourceEvents.ChainPruned, this.handlePruneL2Blocks.bind(this));
```

Comment:
I realise this isn't new code (in this PR) but this `bind` will create a new function instance and won't remove the original.

Reply:
good spot